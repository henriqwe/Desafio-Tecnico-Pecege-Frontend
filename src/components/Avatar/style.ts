import styled from 'styled-components'

const variantOptions = {
  sm: {
    width: '4.5rem',
    height: '4.5rem',
    fontSize: '2rem'
  },
  md: {
    width: '6.5rem',
    height: '6.5rem',
    fontSize: '4rem'
  }
}

export type TVariant = keyof typeof variantOptions

export const SAvatar = styled.div<{ variant: TVariant }>`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgb(30 64 175);
  color: white;
  border-radius: 100%;
  width: 4.5rem;
  height: 4.5rem;
  font-size: 2rem;
  ${({ variant }) => variantOptions[variant]}
`
