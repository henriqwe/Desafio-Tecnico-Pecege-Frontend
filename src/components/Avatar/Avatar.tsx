import { SAvatar } from './style'
import type { TVariant } from './style'

interface props {
  name: string
  variant?: TVariant
}

export function Avatar({ name, variant = 'sm' }: props) {
  const avatarLetters = name?.toUpperCase().split(' ')

  return (
    <SAvatar variant={variant}>
      {avatarLetters?.[0]?.[0]}
      {avatarLetters?.[1]?.[0]}
    </SAvatar>
  )
}
