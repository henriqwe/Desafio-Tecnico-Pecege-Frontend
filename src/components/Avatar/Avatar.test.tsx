import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Avatar } from '&components/Avatar/Avatar'

describe('Avatar', () => {
  it('should render correctly snapshot', () => {
    const { container } = render(<Avatar name="John Doe" />)

    expect(container).toMatchSnapshot()
  })
  it('should render the first name letter and the last second name letter', async () => {
    const { findByTestId } = render(<Avatar name="John Doe" />)

    const avatar = (await findByTestId('avatar-container')) as HTMLDivElement
    expect(avatar.innerHTML).toEqual('JD')
  })
})
