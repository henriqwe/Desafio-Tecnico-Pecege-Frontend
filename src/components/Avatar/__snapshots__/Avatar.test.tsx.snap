// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render correctly snapshot 1`] = `
<div>
  <div
    class="sc-bcPKhP kipEEd"
    data-testid="avatar-container"
    variant="sm"
  >
    J
    D
  </div>
</div>
`;
